modules/osdk-golang-implement-controller.adoc

```diff
@@ -0,0 +1,146 @@
+
+[id="osdk-golang-implement-controller-reconcile-loop_{context}"]
+== Reconcile loop
+
+Every controller has a reconciler object with a `Reconcile()` method that implements the reconcile loop. The reconcile loop is passed the `Request` argument, which is a namespace and name key used to find the primary resource object, `Memcached`, from the cache:
+
+[source,go]
+----
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	cachev1alpha1 "github.com/example-inc/memcached-operator/api/v1alpha1"
+	...
+)
+
+func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+  // Lookup the Memcached instance for this reconcile request
+  memcached := &cachev1alpha1.Memcached{}
+  err := r.Get(ctx, req.NamespacedName, memcached)
+  ...
+}
+----
+
+Based on the return values, result, and error, the request might be requeued and the reconcile loop might be triggered again:
+
+[source,go]
+----
+// Reconcile successful - don't requeue
+return ctrl.Result{}, nil
+// Reconcile failed due to error - requeue
+return ctrl.Result{}, err
+// Requeue for any reason other than an error
+return ctrl.Result{Requeue: true}, nil
+----
+
+You can set the `Result.RequeueAfter` to requeue the request after a grace period as well:
+
+[source,go]
+----
+import "time"
+
+// Reconcile for any reason other than an error after 5 seconds
+return ctrl.Result{RequeueAfter: time.Second*5}, nil
+----
+
+[NOTE]
+====
+You can return `Result` with `RequeueAfter` set to periodically reconcile a CR.
+====
+
+For more on reconcilers, clients, and interacting with resource events, see the link:https://sdk.operatorframework.io/docs/building-operators/golang/references/client/[Controller Runtime Client API] documentation.
+
+[id="osdk-golang-implement-controller-rbac-markers_{context}"]
+== Permissions and RBAC manifests
+
+The controller requires certain RBAC permissions to interact with the resources it manages. These are specified using RBAC markers, such as the following:
+
+[source,go]
+----
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;
+
+func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+  ...
+}
+----
+
+The `ClusterRole` object manifest at `config/rbac/role.yaml` is generated from the above markers using the `controller-gen` utility whenever the `make manifests` command is run.
```

Comment:
> ! above adjective, adverb, preposition
1. Do not use to refer to a subsequent software version or fix level. Use "later". 2. Do not use to indicate a relative location in a document, as in "the above restrictions" or "the steps above". Use "previous" or "preceding". Examples: "The previous restrictions apply to ..."; "The preceding steps..."
```suggestion
The `ClusterRole` object manifest at `config/rbac/role.yaml` is generated from the previous markers by using the `controller-gen` utility whenever the `make manifests` command is run.
```